### PR TITLE
Refactor for scraped

### PR DESCRIPTION
**Pull request by ondenman**
This is part of: https://github.com/everypolitician/everypolitician-data/issues/25302

The next step will be to add a decorator for the image paths. (Issue: https://github.com/everypolitician-scrapers/Alderney/issues/5)

### CODE REVIEW

File: test/member_page_test.rb [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  end
+
+  describe 'Page with image' do
+    let(:url) { 'http://www.alderney.gov.gg/article/4081/Ian-Tugby' }
+
+    it 'should return the expected data' do
+      subject.to_h.must_equal(
+        id:     '4081',
+        name:   'Ian Tugby',
+        image:  './media/image/6/o/140913_Alderney-801.jpg',
```
Comment (tmtmtmtm) [inline test/member_page_test.rb]: As before, you should really add a comment here (ideally with an issue number) to say that you know this is broken, and you're going to fix it RSN™

File: scraper.rb [hunk 2]
```diff
@@ -3,43 +3,29 @@
...
-  # end
-  data[:image] = URI.join(@BASE, URI.escape(data[:image])).to_s unless data[:image].to_s.empty?
-  ScraperWiki.save_sqlite(%i(id term), data)
-end
-
-@BASE = 'http://www.alderney.gov.gg'
-
 ScraperWiki.sqliteexecute('DELETE FROM data') rescue nil
-scrape_list('http://www.alderney.gov.gg/article/4077/States-Members')
+scrape_list(2014, 'http://www.alderney.gov.gg/article/4077/States-Members')
```
Comment (tmtmtmtm) [inline scraper.rb]: There doesn't seem to be much value in having a separate `scrape_list` here any more, especially if you're changing its signature anyway

File: scraper.rb [hunk 3]
```diff
@@ -3,43 +3,29 @@
...
-def scrape_list(url)
-  noko = noko_for(url)
-  noko.xpath('.//ul[@id="leftnavigation"]/li[contains(.,"States Members")]/following-sibling::li[@class="child"]//a/@href').each do |li|
-    scrape_mp(li.text)
+def scrape_list(term, url)
+  (scrape url => MembersPage).member_urls.each do |mem_url|
+    data = (scrape mem_url => MemberPage).to_h.merge(district: 'Alderney',
+                                                     party:    'Independent',
+                                                     term:     term)
+    ScraperWiki.save_sqlite(%i(id term), data)
```
Comment (tmtmtmtm) [inline scraper.rb]: As discussed in Leeds, unless there's a good reason, let's aim for consistency by building up the `data` and then saving it in one shot.

**Review by tmtmtmtm — CHANGES_REQUESTED** [review]

File: test/member_page_test.rb (hunk 1 above)
Comment (ondenman) [reply test/member_page_test.rb]: As a temporary fix, I've gone back to making the URL absolute in the MemberClass. The pre-factor data contains images with absolute URLs. I've noted it in a comment in 2ba6b1fc18d9f74e1b2b1ca02bdd0be374dd8972

**Review by tmtmtmtm — APPROVED** [review]
:+1:

**Pull request merged**